lib/fs.js

```diff
@@ -320,7 +321,7 @@ fs.existsSync = function(path) {
 };
 
 fs.readFile = function(path, options, callback) {
-  callback = maybeCallback(arguments[arguments.length - 1]);
+  callback = maybeCallback(callback || options);
```

Comment:
I see why this might indeed be "semver-major-y" but it looks fine to me. The only difference is in the case callback is passed but not as a function at which point `options` would be passed and would be rethrown if real options passed.

I think a CITGM run would be sufficient here and if that passes I don't think we need to change it or semver-major it.

Also I'm very surprised arguments use only indexed access and length check is performed deopts.